python/paddle/fluid/dygraph/dygraph_to_static/convert_operators.py

```diff
@@ -231,17 +234,30 @@ def _run_paddle_cond(pred, true_fn, false_fn, get_args, set_args,
 
     def new_true_fn():
         set_args(init_args)
-        outs = true_fn()
-        _check_no_undefined_var(outs, return_name_ids, 'if_body')
-        return outs
+        true_fn()
+        return get_args()
 
     def new_false_fn():
         set_args(init_args)
-        outs = false_fn()
-        _check_no_undefined_var(outs, return_name_ids, 'else_body')
-        return outs
-
-    cond_outs = control_flow.cond(pred, new_true_fn, new_false_fn)
+        false_fn()
+        return get_args()
+
+    try:
+        cond_outs = control_flow.cond(pred, new_true_fn, new_false_fn,
+                                      return_name_ids)
+    except Exception as e:
+        if re.search("Unsupported return type of true_fn and false_fn in cond",
+                     str(e)):
+            raise Dygraph2StaticException(
+                "Your if/else have different return type. TODO: add link to modifty."
```

Comment:
```suggestion
              # TODO: add link to modifty.
                "Your if/else has different return type. %s" % e
```
这里如果还没有加TODO的话，最好先把之前的err msg re-throw下